app/src/main/res/xml/preference_files.xml

```diff
@@ -21,6 +21,12 @@
 
     <PreferenceCategory android:title="@string/pref_category_encoding">
 
+        <SwitchPreference
+            android:title="@string/pref_encoding_auto_detect_title"
+            android:summary="@string/pref_encoding_auto_detect_summary"
+            android:defaultValue="false"
+            android:key="ENCODING_AUTO_DETECT"/>
```

Comment:
I think it will be great if we could disable 2 other encoding preferences when this enabled, so, here's the steps:
1) Add `android:disableDependentsState="true"` to `ENCODING_AUTO_DETECT` preference
2) Add `android:dependency="ENCODING_AUTO_DETECT"` to `ENCODING_FOR_OPENING` preference
3) Add `android:dependency="ENCODING_AUTO_DETECT"` to `ENCODING_FOR_SAVING` preference

Reply:
great idea!